sdk/storage/Azure.Storage.DataMovement.Blobs/stress/src/Scenarios/UploadBlobSingleScenarioBase.cs

```diff
@@ -0,0 +1,80 @@
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Specialized;
+using Azure.Storage.DataMovement.Tests;
+using Azure.Storage.Blobs.Models;
+
+namespace Azure.Storage.DataMovement.Blobs.Stress
+{
+    public abstract class UploadBlobSingleScenarioBase : BlobScenarioBase
+    {
+        protected UploadBlobSingleScenarioBase(
+            Uri blobUri,
+            int? blobSize,
+            TransferManagerOptions transferManagerOptions,
+            DataTransferOptions dataTransferOptions,
+            TokenCredential tokenCredential,
+            Metrics metrics,
+            string testRunId) : base(blobUri, blobSize, transferManagerOptions, dataTransferOptions, tokenCredential, metrics, testRunId)
+        {
+        }
+
+        internal async Task RunTestInternalAsync(BlobType blobType, CancellationToken cancellationToken = default)
+        {
+            DisposingLocalDirectory disposingLocalDirectory = DisposingLocalDirectory.GetTestDirectory();
+            string destinationContainerName = TestSetupHelper.Randomize("container");
+            BlobContainerClient destinationContainerClient = _blobServiceClient.GetBlobContainerClient(destinationContainerName);
+            await destinationContainerClient.CreateIfNotExistsAsync();
+            string blobName = TestSetupHelper.Randomize("blob");
+
+            // Create Local Source Storage Resource
+            StorageResource sourceResource = await TestSetupHelper.GetTemporaryFileStorageResourceAsync(
+                disposingLocalDirectory.DirectoryPath,
+                fileName: blobName,
+                fileSize: _blobSize,
+                cancellationToken: cancellationToken);
+
+            // Create Destination Storage Resource
+            BlobBaseClient destinationBaseBlob;
+            StorageResource destinationResource;
+            if (blobType == BlobType.Append)
+            {
+                AppendBlobClient destinationBlob = destinationContainerClient.GetAppendBlobClient(blobName);
+                destinationBaseBlob = destinationBlob;
+                destinationResource = _blobsStorageResourceProvider.FromClient(destinationBlob);
+            }
+            else if (blobType == BlobType.Page)
+            {
+                PageBlobClient destinationBlob = destinationContainerClient.GetPageBlobClient(blobName);
+                destinationBaseBlob = destinationBlob;
+                destinationResource = _blobsStorageResourceProvider.FromClient(destinationBlob);
+            }
+            else
+            {
+                BlockBlobClient destinationBlob = destinationContainerClient.GetBlockBlobClient(blobName);
+                destinationBaseBlob = destinationBlob;
+                destinationResource = _blobsStorageResourceProvider.FromClient(destinationBlob);
+            }
+
+            // Start Transfer
+            await new TransferValidator()
+            {
+                TransferManager = new()//new(_transferManagerOptions)
```

Comment:
Did you actually want to pass the options here? And below for `_dataTransferOptions`.

Reply:
yeah. I was testing something out and found a bug someplace else. added these options bag back in